### PR TITLE
MakeMeASandwich: Switching to template

### PR DESCRIPTION
**Pull request by mintsoft**
###### Description of changes

Ref: https://github.com/duckduckgo/zeroclickinfo-goodies/issues/2767

###### People to notify (@mention interested parties)

---

Instant Answer Page: https://duck.co/ia/view/make_me_asandwich

[Maintainer](http://docs.duckduckhack.com/maintaining/guidelines.html): @mattr555

### CODE REVIEW

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

**Comment by mintsoft** [comment]
This one doesn't currently render in `duckpan server` something is wrong but I can't see it atm

**Comment by GuiltyDolphin** [comment]
@mintsoft Nothing jumps out at me - though I haven't scrutinized the handlebars yet.

**Comment by mintsoft** [comment]
@GuiltyDolphin Figured it out I think; the name of the directory in share needs to be the same as the id of the IA; NOT the type of the IA returned. I have no idea why we have both; this is just unnecessarily confusing :)

**Comment by GuiltyDolphin** [comment]
@mintsoft Ah! Yeah - everything with underscores is usually just the ID - which can sometimes differ quite a bit from the camel name!

**Comment by mintsoft** [comment]
I always thought it was the same as the `zci answer_type => 'xkcd_sandwich';` this must be the first situation where it's been different that I've come across

**Comment by GuiltyDolphin** [comment]
@mintsoft Uhh.. What even is that `answer_type` for...? I thought it was for the ID, in which case we can surely have it implicit...?

**Comment by mintsoft** [comment]
@GuiltyDolphin pre-metadata stuff it was used as the id of the IA; the tests used to use it to assert that the right IA is returned. I'm guessing it might be used to link the IA source to the IA metadata now?

**Comment by GuiltyDolphin** [comment]
@mintsoft Uhh... I know it ain't used by DuckPAN... And it doesn't look like DDG uses it either; platform doesn't even reference it...

**Comment by moollaza** [comment]
Deploying to beta...

File: t/MakeMeASandwich.t [hunk 1]
```diff
@@ -10,27 +10,32 @@ use DDG::Test::Goodie;
...
+{
+    my ($text, $input) = @_;
+    return test_zci($text, structured_answer => {
+        data => {
+            result => $text,
+            input => $input
+        },
+        templates => {
+            options => {
+                content => 'DDH.make_me_asandwich.content'
```
Comment (moollaza) [inline t/MakeMeASandwich.t]: We could abandon the HTML if we make the subtitle an array:

```
subtitle = [
 { text: $input },
 { text: "XKCD 149", href: "https://duckduckgo.com/?q=xkcd%20149" }
]
```

Would look like:

> **What? Make it yourself.**
> "Make me a sandwich" | XKCD 149

File: share/goodie/make_me_asandwich/content.handlebars [hunk 2]
```diff
@@ -0,0 +1,6 @@
+<div class="zci__body">
+    <h3 class="zci__caption  zci__result">{{result}}</h3>
```
Comment (moollaza) [inline share/goodie/make_me_asandwich/content.handlebars]: using a `title` here should work OK

**Comment by mintsoft** [comment]
@moollaza I've switched that but I'm getting weird display in duckpan now:

duckpannage

**Comment by mintsoft** [comment]
@moollaza Never mind; sorted it!

**Comment by mattr555** [comment]
LGTM, merging :+1:

**Comment by mintsoft** [comment]
Thanks @mattr555

**Pull request merged**